src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,48 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
```

Comment:
```suggestion
public class MachineServiceImpl implements MachineService<Machine> {
```
в інтерфейсі ми використовуємо абстрактний T щоб було більше гнучкості для імплементації так як замість нього можна підставити різні типи (Machine / Bulldozer ..). Тому тут вже потрібно підставити конкретику і це накладе обмеження на методи щоб не можна було там інші типи використати